src/jit/lower.cpp

```diff
@@ -3953,6 +4025,16 @@ void Lowering::DoPhase()
         currBlock = block;
         comp->compCurBB = block;
 
+#if !defined(_TARGET_64BIT_)
+        // Decompose TYP_LONG nodes
+        for (stmt = block->bbTreeList; stmt != nullptr; stmt = stmt->gtNext)
+        {
+            comp->compCurStmt = stmt;
+            comp->fgWalkTreePost(&stmt->gtStmt.gtStmtExpr, &Lowering::DecompNodeHelper, this, true);
+            stmt = comp->compCurStmt;
+        }
+#endif
+
         /* Walk the statement trees in this basic block */
```

Comment:
Yes, embedded statements created by `fgInsertEmbeddedFormTemp` aren't lowered without this change.

But as mentioned in another issue I'm wondering if decomposition shouldn't happen after lowering. That may simplify GT_IND handling because after lowering we have GT_LEA nodes.